### ISSUE TITLE
I.L. FE: H1 should be included in <title>

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#14046]
NOT LAUNCH-BLOCKING FOR MVP = post-mvp

## Description
The page title (`<title>`) for each page in the Income Limits app should be `{H1} | Income Limits | Veterans Affairs`. Currently the H1 is missing from the beginning. 

Adding the H1 in this way will:
- Orient the user to where in the app there are
- Correct the current WCAG violation (all pages must be uniquely titled)

### User story

**AS A** Veteran 
**I WANT** the H1 to be included in the <title> tag
**SO THAT** I can differentiate which page I'm on from other pages in the app

### Engineering notes / background
We can set the `document.title` in a `useEffect` on every Income Limits page. Ideally we create a utility that can be reused to add `| Income Limits | Veterans Affairs` to the `<h1>` on each page.

## Acceptance criteria

- [x] Confirmed whether the Design System team has solved this for us before we took it on
- [x] Each page in the Income limits app has a unique page title and follow the pattern `{H1} | Income Limits | Veterans Affairs`
- [ ] Accessibility review

**Comment by wesrowe** [comment]
@laflannery, your pattern suggests the <title> of the page would be a question. Is that what you would expect for a form step screen?

**Comment by laflannery** [comment]
Yes @wesrowe , that is what we are expecting here

**Comment by randimays** [comment]
On initial pass, pointed this as a 5 tentatively, but I think we should think though some technical details in team refinement because the path forward isn't clear for me yet. @dsasser @jtmst please weigh in if you have a chance to review this. CC: @wesrowe @jilladams

Income Limits is the first application using the subtask pattern (one question per page). All React apps are configured in `content-build` in the `registry.json` [here](https://github.com/department-of-veterans-affairs/content-build/blob/524837be536013fc83717950408e1776d91ab713/src/applications/registry.json#L1391). The `title` attribute from this configuration is plugged into the page metadata [here](https://github.com/department-of-veterans-affairs/content-build/blob/524837be536013fc83717950408e1776d91ab713/src/site/includes/metatags.drupal.liquid#L51). 

Considerations:
- `content-build` has no knowledge of Income Limits questions to plug in to the title; these live in `vets-website`
- There is no existing logic in `content-build` to apply page titles conditionally; the app title is applied for all pages
- Does not look like there's an existing pattern of setting page titles from `vets-website`

Ideas:
1. Add routes/questions to the `registry.json` for the Income Limits application in `content-build` and adjust the logic consuming this data to account for it. **Unknown**: how to conditionally apply title based on current route
2. Manually override page titles in `vets-website` for each page. Likely simple, but not a great long-term solution

**Comment by laflannery** [comment]
@wesrowe and @randimays Just wanted to document here something that I had put in slack:
- Because this is a defect-2 it would not be launch blocking
- It should ideally be resolved relatively soon post-launch though

Just keep me in the loop regarding what you find on the technical feasibility side of things so I am aware of where we stand and what our plan is, thanks!

**Comment by wesrowe** [comment]
@randimays, thanks for the thorough consideration. I think it's largely an engineering question how to do this in a way that meets a11y criteria and conforms to how Platform would want to do it. I think you could either 1) make interacting with Platform part of this ticket, or 2) request a spike ticket. 1 is probably easier, you can pad the estimate to 8 if necessary.

**Comment by wesrowe** [comment]
@randimays, technical clarification - you said "Income Limits is the first application using the subtask pattern." I had been led to believe that the subtask pattern is commonly used on forms. Is that not the case?

**Comment by randimays** [comment]
@wesrowe I checked with Laura to see if she knew of another form that was multi-page like this one and she said she doesn't know of any that are using this complete subtask pattern.
https://dsva.slack.com/archives/C52CL1PKQ/p1686771443782199?thread_ts=1686770901.155189&cid=C52CL1PKQ

**Comment by randimays** [comment]
@wesrowe Another clarification after digging a bit deeper. There are other forms out there using the subtask pattern:
https://staging.va.gov/education/apply-for-education-benefits/application/1990/benefits-eligibility/benefits-selection
https://staging.va.gov/decision-reviews/supplemental-claim/file-supplemental-claim-form-20-0995/introduction

But both forms (and likely others) have the same `<h1>` element across every page, so the `<title>` element does not need to change.

**Comment by laflannery** [comment]
This is the [current existing issue](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1326) in the DST repo regarding the subtask pattern. It includes the H1s but also quite a few other suggestions and experimental changes.

**Comment by randimays** [comment]
@laflannery The current existing issue doesn't seem to specifically address the page titles being dynamic, but if we're following accessibility guidelines close with a redesign as proposed, that would naturally happen, right? Wanting to know if it's something we should call out explicitly on the ticket or when checking in with them when our ticket comes up.

**Comment by wesrowe** [comment]
Note: there is much intra-team discussion in [this slack thread](https://dsva.slack.com/archives/C52CL1PKQ/p1686770901155189).

**Comment by laflannery** [comment]
@randimays I added a comment to that ticket

**Comment by wesrowe** [comment]
Remind me @randimays, was this ticket deemed not possible at this time?

**Comment by randimays** [comment]
I talked this over with @chriskim2311 and we think a spike might be helpful here. We need to engage with Platform and see if they have any suggestions about how best to tackle this.

I'm not certain where the ownership is here; whether Platform needs to make these types of structural changes or if we can take the initiative to add something in that we vet with them first.

The only solution we could come up with that doesn't involve a conversation with Platform was to manually inject a new `<title>` in `vets-website`, but this would be messy and likely not address our accessibility issues.

**Comment by randimays** [comment]
@wesrowe I'm not sure exactly who would be making the structural changes here. Maybe the design system team?

**Comment by wesrowe** [comment]
@randimays, I think it makes sense for you to reach out to the DST and inquire about steps forward. Do they have an issue for it yet? If not, maybe that's a place to start. (I learned this from you actually. :) )

**Comment by randimays** [comment]
@wesrowe Done https://dsva.slack.com/archives/C01DBGX4P45/p1690995889915049

**Comment by randimays** [comment]
@wesrowe I built a POC branch for simply changing the `<title>` in vets-website after the page loads. I sent the POC over to @laflannery to do a quick accessibility check, and she agrees that this is a good solution.

I'll re-point this with that in mind and add some detail to the Engineering Notes.

Slack conversation: https://dsva.slack.com/archives/C52CL1PKQ/p1691599666992189

**Comment by wesrowe** [comment]
If it works for Laura it works for me :)

**Comment by randimays** [comment]
@laflannery and I reviewed in staging. This is ready to close.

Slack thread: https://dsva.slack.com/archives/C52CL1PKQ/p1692299329826309?thread_ts=1692284261.335099&cid=C52CL1PKQ
